controller_manager/src/controller_manager.cpp

```diff
@@ -878,7 +879,8 @@ controller_interface::return_type ControllerManager::switch_controller(
     return controller_interface::return_type::ERROR;
   }
 
-  switch_params_ = SwitchParams();
+  // reset the switch param internal variables
```

Comment:
is there a specific reason for changing from a "stateless" reinitialization setup to this more stateful one? The constructor did the same job...

Reply:
you are right! I had to do it because of the internal mutex and conditional variable

Reply:
I find this good. Controller switching is “stateful” per definition, so I think it is OK for keep it.

Reply:
Very valid, thanks for the explanation!